.github/workflows/docs.yaml

```diff
@@ -25,21 +25,28 @@ jobs:
     - name: Checkout Code
       uses: actions/checkout@v4
 
-    - name: Set up Python 3.10
-      uses: actions/setup-python@v5
-      with:
-        python-version: "3.10"
+    - name: Set up Environment
+      id: setup
+      run: |
+        curl -LsSf https://astral.sh/uv/install.sh | sh
+        uv venv
+        echo "VIRTUAL_ENV=$HOME/.venv" >> $GITHUB_ENV
+      continue-on-error: false
+
+    - name: Install Docs Dependencies
+      id: install_build_dependencies
+      run: |
+        . .venv/bin/activate
+        uv pip install "lm-buddy[docs] @ ."
```

Comment:
We pay a bit of a time cost here because `lm-buddy[docs]` also installs the base package. But it keeps the versions in sync with `pyproject.toml` as the source of truth.

Another approach could be to write out a `requirements-docs.txt` file in the repo and use dynamic metadata to pull that into `pyproject.toml`.

Reply:
this is related to the pypa/pip issue you shared in slack?

Reply:
It is, yeah. There is no way I can find to install only the extras. 

Same goes for Ruff, and how you tried to parse the toml section to get around that.

Using pyproject alone without requirements files is lacking here a bit.